astrolabe/runner.py

```diff
@@ -223,17 +267,63 @@ def run(self, persist_cluster=False, startup_time=1):
                 self.cluster_name))
 
         return junit_test
+        
+    def wait_for_idle(self):
+        selector = BooleanCallablePoller(
+            frequency=self.config.polling_frequency,
+            timeout=self.config.polling_timeout)
+        LOGGER.info("Waiting for cluster maintenance to complete")
+        selector.poll([self], attribute="is_cluster_state", args=("IDLE",),
+                      kwargs={})
+                      
+    def retrieve_logs(self):
+        data = self.admin_client.request('GET', '/api/private/nds/groups/%s/clusters/%s' % (self.project.id, self.cluster_name)).data
```

Comment:
After rebasing on #99, and reverting the changes made in this PR to `AtlasClient.construct_resource_url`, this should simply be:

```
data = self.admin_client.nds.groups[self.project.id].clusters[self.cluster_name].get(api_version='private').data
```

Reply:
Changed!